packages/react-shopify-app-bridge/src/Provider.tsx

```diff
@@ -161,27 +147,66 @@ const InnerGadgetProvider = memo(
   }
 );
 
+const StandaloneInnerProvider = ({
+  isRootFrameRequest,
+  children,
+  query,
+  gadgetAppUrl,
+  type,
+}: {
+  isRootFrameRequest: boolean;
+  children: ReactNode;
+  query?: URLSearchParams;
+  gadgetAppUrl: string;
+  type: AppType;
+}) => {
+  // we still need to run the oauth process if we're in an install context so we should redirect to gadget
+  const isInstallRequest = query?.has("hmac") && query?.has("shop");
+  const runningShopifyAuth = isInstallRequest;
+
+  // redirect to Gadget to initiate the oauth process if we need to.
+  useEffect(() => {
+    if (!runningShopifyAuth || isRootFrameRequest) return;
+    // redirect to gadget app root pages url with oauth params
+    const redirectURL = new URL("/api/connections/auth/shopify", gadgetAppUrl);
+    redirectURL.search = query?.toString() ?? "";
+    const redirectURLWithOAuthParams = redirectURL.toString();
+
+    console.debug("[gadget-rsab] redirecting to gadget to initiate oauth process in standalone mode or missing app bridge global", {
+      appType: type,
+      isInstallRequest,
+      redirectURL: redirectURLWithOAuthParams,
+      gadgetAppUrl,
+    });
+
+    window.location.assign(redirectURLWithOAuthParams);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [gadgetAppUrl, isRootFrameRequest, query, runningShopifyAuth]);
+
+  return (
+    <GadgetAuthContext.Provider
+      value={{
+        isAuthenticated: false,
+        isEmbedded: false,
+        canAuth: false,
+        loading: false,
+        appBridge: null,
+        isRootFrameRequest: isRootFrameRequest,
+      }}
+    >
+      {children}
+    </GadgetAuthContext.Provider>
+  );
+};
```

Comment:
Extracting this was a great call 👍 We could potentially make it a different top level export so that you use a different provider from our package when embedded vs standalone, and then it would treeshake out a bit nicer. Don't have to make this change rn but something to aim for long term methinks